das/retry_wrapper.go

```diff
@@ -0,0 +1,64 @@
+// Copyright 2021-2022, Offchain Labs, Inc.
+// For license information, see https://github.com/nitro/blob/master/LICENSE
+
+package das
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/cenkalti/backoff/v4"
+
+	"github.com/offchainlabs/nitro/arbstate"
+)
+
+type RetryWrapper struct {
+	backoffPolicy backoff.BackOff
+	DataAvailabilityService
+}
+
+func NewRetryWrapper(dataAvailabilityService DataAvailabilityService) DataAvailabilityService {
+	backoffPolicy := backoff.NewExponentialBackOff()
+	backoffPolicy.InitialInterval = time.Millisecond * 100
+	return &RetryWrapper{
+		backoffPolicy:           backoffPolicy,
+		DataAvailabilityService: dataAvailabilityService,
+	}
+}
+
+func (w *RetryWrapper) GetByHash(ctx context.Context, hash []byte) ([]byte, error) {
+	var res []byte
+	err := backoff.Retry(func() error {
+		data, err := w.DataAvailabilityService.GetByHash(ctx, hash)
```

Comment:
If `ctx` is cancelled, we don't want to keep retrying. I think we need to check for one of the context-ending error types, and if we see one of those then wrap it in the backoff package's `Permanent` error wrapper, so the backoff package stops retrying.

Reply:
Done.